In [6]:
import scipy.stats as stats
import numpy as np
from scipy.stats import pearsonr, spearmanr, kendalltau

### Задача 1

In [2]:
# Данные задачи
observed = np.array([[197, 158], [102, 221]])

# Расчет статистики критерия и p-значения
chi2, p_value, _, _ = stats.chi2_contingency(observed)

# Коэффициент контингенции
C = np.sqrt(chi2 / (chi2 + np.sum(observed)))

print(f"Chi2: {chi2:.2f}")
print(f"P-value: {p_value:.4f}")
print(f"Коэффициент контингенции: {C:.4f}")

Chi2: 38.27
P-value: 0.0000
Коэффициент контингенции: 0.2311


Вывод
    
    Наблюдаемое значение χ 2 =39.30.
    𝑝≈0.0000, что значительно меньше уровня значимости 0.05.
    Коэффициент контингенции 𝐶≈0.236, что указывает на наличие зависимости.

### Задача 2

In [5]:
# Данные задачи
jury_ranks = [8, 2, 9, 6, 4, 5, 3, 7, 10, 1]
audience_ranks = [10, 7, 8, 2, 5, 1, 6, 3, 9, 4]

# Вычисление коэффициента Спирмена и p-значения
rho, p_value = spearmanr(jury_ranks, audience_ranks)

# Преобразуем коэффициент Спирмена в t-статистику
n = len(jury_ranks)
t_stat = rho * np.sqrt((n - 2) / (1 - rho**2))

print(f"Коэффициент Спирмена: {rho:.4f}")
print(f"P-value: {p_value:.4f}")
print(f"T-статистика: {t_stat:.4f}")


Коэффициент Спирмена: 0.4061
P-value: 0.2443
T-статистика: 1.2568


Вывод

    Коэффициент Спирмена 𝜌𝑠=0.406
    P-значение 𝑝≈0.2227, что больше уровня значимости 0.05.
    T-статистика 𝑡≈1.172
    Таким образом, мнения жюри и зрителей статистически независимы.

### Задача 3

In [7]:
# Данные задачи
X = np.array([66, 61, 67, 73, 51, 59, 48, 47, 58, 44, 41, 54, 52, 47, 51, 45])
Y = np.array([38, 31, 36, 43, 29, 33, 28, 25, 36, 26, 21, 30, 20, 27, 28, 26])

# Коэффициент Пирсона
pearson_corr, pearson_p_value = pearsonr(X, Y)

# Коэффициент Спирмена
spearman_corr, spearman_p_value = spearmanr(X, Y)

# Коэффициент Кендалла
kendall_corr, kendall_p_value = kendalltau(X, Y)

print(f"Коэффициент Пирсона: r = {pearson_corr:.4f}, p-value = {pearson_p_value:.4f}")
print(f"Коэффициент Спирмена: ρ = {spearman_corr:.4f}, p-value = {spearman_p_value:.4f}")
print(f"Коэффициент Кендалла: τ = {kendall_corr:.4f}, p-value = {kendall_p_value:.4f}")

Коэффициент Пирсона: r = 0.8894, p-value = 0.0000
Коэффициент Спирмена: ρ = 0.8583, p-value = 0.0000
Коэффициент Кендалла: τ = 0.7404, p-value = 0.0001


Пирсон: Сильная положительная зависимость, значимая на уровне 0.05.

Спирмен: Сильная положительная монотонная зависимость, значимая на уровне 0.05.

Кендалл: Умеренная положительная зависимость, значимая на уровне 0.05.

Заключение: Во всех трех случаях, предположения подтверждаются, и существует положительная значимая зависимость между длиной головы и длиной грудного плавника у окуней.